vllm/engine/output_processor/multi_step.py

```diff
@@ -0,0 +1,117 @@
+from typing import Callable, Iterable, List
```

Comment:
Another question:
From my understanding, multi_step processor is used for target model, which is not a muti_step_worker.
single_step processor is used for the draft model, which is a multi_step_worker. 
Correct me if I'm wrong. If this is the case, can we have a better name for multi_step output processor?

Reply:
The LLMEngine only interacts with the SpecDecodeWorker, which internally manages workers for the draft and target models. See this diagram from the meetup below:

![Screenshot 2024-04-10 at 2 09 30 PM](https://github.com/vllm-project/vllm/assets/950914/6b3f62ed-f100-40d9-ba0e-682f6b189fa8)

This means that from the LLMEngine's perspective, there is only one worker type, and only one output processor type at a time.